srml/staking/src/lib.rs

```diff
@@ -658,6 +668,9 @@ impl<T: Trait> Module<T> {
 			}
 		}
 
+		// prune validators who haven't re-registered after timeout.
+		Self::prune_timed_out();
```

Comment:
i'd prefer not to have any non-critical additional unbounded linked list iteration.